frontend/pages/policies/ManagePoliciesPage/components/ManageAutomationsModal/ManageAutomationsModal.tsx

```diff
@@ -0,0 +1,230 @@
+  currentDestinationUrl?: string;
+}
+
+interface ICheckedPolicy {
+  name?: string;
+  id: number;
+  isChecked: boolean;
+}
+
+const useCheckboxListStateManagement = (
+  allPolicies: IPolicy[],
+  automatedPolicies: number[] | undefined
+) => {
+  const [policyItems, setPolicyItems] = useState<ICheckedPolicy[]>(() => {
+    return (
+      allPolicies &&
+      allPolicies.map((policy) => {
+        return {
+          name: policy.name,
+          id: policy.id,
+          isChecked:
+            !!automatedPolicies && automatedPolicies.includes(policy.id),
+        };
+      })
+    );
+  });
+
+  const updatePolicyItems = (policyId: number) => {
+    setPolicyItems((prevState) => {
+      const selectedPolicy = policyItems.find(
+        (policy) => policy.id === policyId
+      );
+
+      const updatedPolicy = selectedPolicy && {
+        ...selectedPolicy,
+        isChecked: !!selectedPolicy && !selectedPolicy.isChecked,
+      };
+
+      // this is replacing the policy object with the updatedPolicy we just created.
+      const newState = prevState.map((currentPolicy) => {
+        return currentPolicy.id === policyId && updatedPolicy
+          ? updatedPolicy
+          : currentPolicy;
+      });
+      return newState;
+    });
+  };
+
+  return { policyItems, updatePolicyItems };
+};
+
+const validateAutomationURL = (url: string) => {
+  const errors: { [key: string]: string } = {};
+
+  if (!url) {
+    errors.url = "Please add a valid destination URL";
+  }
+
+  const valid = !size(errors);
+  return { valid, errors };
+};
+
+const baseClass = "manage-automations-modal";
+
+const ManageAutomationsModal = ({
+  onCancel: onReturnToApp,
+  onCreateAutomationsSubmit,
+  togglePreviewPayloadModal,
+  showPreviewPayloadModal,
+  availablePolicies,
+  currentAutomatedPolicies,
+  currentDestinationUrl,
+}: IManageAutomationsModalProps): JSX.Element => {
+  const [destination_url, setDestinationUrl] = useState<string>(
+    currentDestinationUrl || ""
+  );
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
+
+  const { policyItems, updatePolicyItems } = useCheckboxListStateManagement(
+    availablePolicies,
+    currentAutomatedPolicies
+  );
+
+  useDeepEffect(() => {
+    if (destination_url) {
+      setErrors({});
+    }
+  }, [destination_url]);
+
+  const onURLChange = (value: string) => {
+    setDestinationUrl(value);
+  };
+
+  const handleSaveAutomation = (evt: React.MouseEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+
+    const { valid, errors: newErrors } = validateAutomationURL(destination_url);
+    setErrors({
+      ...errors,
+      ...newErrors,
+    });
+
+    if (valid) {
+      const policy_ids =
+        policyItems &&
+        policyItems
+          .filter((policy) => policy.isChecked)
+          .map((policy) => policy.id);
+      const enable_failing_policies_webhook = true; // Leave nearest component in case we decide to add disabling as a UI feature
```

Comment:
I'm not sure what "Leave nearest component" means in the comment.

Reply:
Instead of setting it in the parent component, submit function, or API call since Noah said we'll probably iterate on the UI to have an option to disable.